pkg/queue/concurrency_state.go

```diff
@@ -81,10 +81,43 @@ func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, r
 			return
 		}
 
-		resume()
+		logger.Info("Requests increased from zero")
+		if err := resume(); err != nil {
+			logger.Errorf("Error handling resume request: %v", err)
+			panic(err)
+		}
 		paused = false
+		logger.Debug("From-Zero request successfully processed")
 		mux.Unlock()
 
 		h.ServeHTTP(w, r)
 	}
 }
+
+// ConcurrencyState Request sends to the concurrency state endpoint
+func ConcurrencyStateRequest(endpoint string, action ConcurrencyStateMessageBody) (func() error, error) {
+	bodyText, err := json.Marshal(action)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create message body: %w", err)
+	}
+	return func() error {
+		body := bytes.NewBuffer(bodyText)
+		req, err := http.NewRequest(http.MethodPost, endpoint, body)
+		if err != nil {
+			return fmt.Errorf("unable to create request: %w", err)
+		}
+		req.Header.Add("Token", "nil") // TODO: use serviceaccountToken from projected volume
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return fmt.Errorf("unable to post request: %w", err)
+		}
+		if resp.StatusCode != http.StatusOK {
+			return fmt.Errorf("expected 200 response, got: %d: %s", resp.StatusCode, resp.Status)
+		}
+		return nil
+	}, nil
+}
+
+type ConcurrencyStateMessageBody struct {
```

Comment:
total nit, but I don't think the `Body` suffix is useful here. There's no metadata section or anything going on, so just `ConcurrencyStateMessage` seems fine.

Reply:
the actually see below for a suggestion involving just dropping this struct anyway